### PR TITLE
apinger: add rrd graph support

### PR DESCRIPTION
**Pull request by jempatel**
- add rrdtool1 and rrdcgi1 dependency for RRD graphs
- add RPC to get an overview and update graphs
- fix interface hotplug to restart apinger instance
- add patch to split alarms list in the status

Signed-off-by: Jaymin Patel <jem.patel@gmail.com>

Maintainer: @samm-git @feckert 
Run tested: x86-64, aarch64, virtual box, marvell_armada-3720-db, v21.02.3

Description:
Add RRD configuration support in UCI
Apinger already has built-in support for RRD, expose it to UCI to generate RRD files and graphs.

### CODE REVIEW

**Comment by samm-git** [comment]
Looks amazing. My only concern is hard dependency on rrd, so on minimal setups this would increase significantly. Should we have a subpackage instead?

**Comment by jempatel** [comment]
> Looks amazing. My only concern is hard dependency on rrd, so on minimal setups this would increase significantly. Should we have a subpackage instead?

Added another package apinger-rrd, dependencies are also handled in luci-app-apinger.
https://github.com/openwrt/luci/pull/5887

**Review by feckert — CHANGES_REQUESTED** [review]

File: net/apinger/files/apinger.init [hunk 1]
```diff
@@ -69,6 +72,7 @@ append_target() {
 	[ -n "$avg_loss_samples" ]        && append_config_line "avg_loss_samples ${avg_loss_samples}"
 	[ -n "$avg_loss_delay_samples" ]  && append_config_line "avg_loss_delay_samples ${avg_loss_delay_samples}"
 	[ -n "$alarms" ]                  && append_config_line "alarms override ${alarms}"
+	[ "$rrd" == "1" ]                 && append_config_line "rrd file \"$APINGER_RRD/apinger-target-$target.rrd\""
```
[the remark this answers is not in the document]
Comment (jempatel) [reply net/apinger/files/apinger.init]: Updated

File: net/apinger/Makefile [hunk 2]
```diff
@@ -65,7 +65,27 @@ define Package/apinger/install
...
+  URL:=https://github.com/Jajcus/apinger
+endef
+
+define Package/apinger-rrd/description
+	Generate RRD Graphs from Apinger Data
+endef
+
+define Package/apinger-rrd/install
+	$(INSTALL_DIR) $(1)/lib/apinger/rpc
+	$(INSTALL_DATA) ./files/graphs.sh $(1)/lib/apinger/rpc
```
[the remark this answers is not in the document]
Comment (jempatel) [reply net/apinger/Makefile]: Updated

**Comment by feckert** [comment]
looks good

**Pull request merged**